serial_posix.go

```diff
@@ -112,6 +112,9 @@ func (p *port) Read(b []byte) (n int, err error) {
 		return
 	}
 	n, err = syscall.Read(fd, b)
+	if n == 0 {
```

Comment:
I would maybe check the orignal error if  set, to not override it.
```suggestion
	if n == 0 && err == nil {
```

Reply:
@UzixLS what do you think?